contracts/src/v0.8/functions/tests/1_0_0/FunctionsRouter.t.sol

```diff
@@ -3,116 +3,1346 @@ pragma solidity ^0.8.19;
+      requestData,
+      FunctionsRequest.REQUEST_DATA_VERSION,
+      500_000,
+      s_donId
+    );
+  }
+
+  function test_SendRequest_RevertIfEmptyData() public {
+    // Build invalid request data
+    bytes memory emptyRequestData = new bytes(0);
+
+    vm.expectRevert(FunctionsRouter.EmptyRequestData.selector);
+    s_functionsRouter.sendRequest(
+      s_subscriptionId,
+      emptyRequestData,
+      FunctionsRequest.REQUEST_DATA_VERSION,
+      5_000,
+      s_donId
+    );
+  }
+
+  event RequestStart(
+    bytes32 indexed requestId,
+    bytes32 indexed donId,
+    uint64 indexed subscriptionId,
+    address subscriptionOwner,
+    address requestingContract,
+    address requestInitiator,
+    bytes data,
+    uint16 dataVersion,
+    uint32 callbackGasLimit,
+    uint96 estimatedTotalCostJuels
+  );
+
+  function test_SendRequest_Success() public {
+    // Build minimal valid request data
+    string memory sourceCode = "return 'hello world';";
+    FunctionsRequest.Request memory request;
+    FunctionsRequest.initializeRequest(
+      request,
+      FunctionsRequest.Location.Inline,
+      FunctionsRequest.CodeLanguage.JavaScript,
+      sourceCode
+    );
+    bytes memory requestData = FunctionsRequest.encodeCBOR(request);
+
+    uint32 callbackGasLimit = 5000;
+
+    bytes32 expectedRequestId = keccak256(
+      abi.encode(address(s_functionsCoordinator), OWNER_ADDRESS, s_subscriptionId, 1)
+    );
+
+    uint96 costEstimate = s_functionsCoordinator.estimateCost(
+      s_subscriptionId,
+      requestData,
+      callbackGasLimit,
+      tx.gasprice
+    );
+
+    vm.recordLogs();
+
+    // topic0 (function signature, always checked), topic1 (true), topic2 (true), topic3 (true), and data (true).
+    bool checkTopic1 = true;
+    bool checkTopic2 = true;
+    bool checkTopic3 = true;
+    bool checkData = true;
+    vm.expectEmit(checkTopic1, checkTopic2, checkTopic3, checkData);
+    emit RequestStart({
+      requestId: expectedRequestId,
+      donId: s_donId,
+      subscriptionId: s_subscriptionId,
+      subscriptionOwner: OWNER_ADDRESS,
+      requestingContract: OWNER_ADDRESS,
+      requestInitiator: OWNER_ADDRESS,
+      data: requestData,
+      dataVersion: FunctionsRequest.REQUEST_DATA_VERSION,
+      callbackGasLimit: callbackGasLimit,
+      estimatedTotalCostJuels: costEstimate
+    });
+
+    bytes32 requestIdFromReturn = s_functionsRouter.sendRequest(
+      s_subscriptionId,
+      requestData,
+      FunctionsRequest.REQUEST_DATA_VERSION,
+      callbackGasLimit,
+      s_donId
+    );
+
+    // Get requestId from RequestStart event log topic 1
+    Vm.Log[] memory entries = vm.getRecordedLogs();
+    bytes32 requestIdFromEvent = entries[2].topics[1];
+
+    assertEq(requestIdFromReturn, requestIdFromEvent);
+  }
 }
 
 /// @notice #sendRequestToProposed
-contract FunctionsRouter_SendRequestToProposed {
+contract FunctionsRouter_SendRequestToProposed is FunctionsSubscriptionSetup {
```

Comment:
The additional test cases I mentioned above can be added for sendRequestToProposed too.

Reply:
Added